src/context/api-context.js

```diff
@@ -0,0 +1,18 @@
+import { createContext, useContext, useReducer } from "react";
+import { initialState, APIReducer } from "../reducers/apiReducer";
+
+const APIContext = createContext();
+
+const APIProvider = ({ children }) => {
+  const [APIState, APIDispatch] = useReducer(APIReducer, initialState);
+
+  return (
+    <APIContext.Provider value={{ APIState, APIDispatch }}>
+      {children}
+    </APIContext.Provider>
+  );
+};
+
+const useAPI = () => useContext(APIContext);
+
+export { APIProvider, useAPI };
```

Comment:
```CategoryContext, categoryState, categoryDispatch``` will be good variable names here. In the future, you will have more API calls so it will conflict then.

Reply:
It might conflict ? I didn't get this part ?